src/components/AddItem.js

```diff
@@ -1,11 +1,102 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { db } from '../lib/firebase';
+
+const initialFormState = {
+  itemName: '',
+  timeFrame: 7,
+  lastPurchased: null,
+};
 
 const AddItem = () => {
+  const token = 'user token';
+  const history = useHistory();
+  const [formData, setFormData] = useState(initialFormState);
+  const [error, setError] = useState('');
+  console.log(error);
+
+  const handleFormChange = (event) => {
+    event.preventDefault();
+    setFormData({
+      ...formData,
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (!formData.itemName) {
+      setError(`${event.target.name} field cannot be empty`);
+    } else {
+      db.collection(token).add(formData);
+      history.push('/list');
+      setFormData(initialFormState);
+      setError('');
+    }
+  };
+
   return (
     <div>
       <h1>Add Item</h1>
+      <form onSubmit={handleSubmit}>
```

Comment:
Nice job with the HTML form! form can be super complicated and it took me a while to realize/understand the full power of the form. Great job!

Reply:
Agree with @MikeBLambert about forms. I thought they were kind of boring before doing a project on that went in depth with forms. It made me notice when forms are difficult to use vs when they are easy to use.

Reply:
Yeah, for more complex forms there are actually entire tools built around handling form state. https://formik.org/ is one of the more popular ones. By no means am I suggesting using it here, but just I just wanted to share it in case you find yourself struggling with a complex form in a future project.